### ISSUE TITLE
Use SVG files that can be styled instead of inline data SVGs for Link component

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#7542]
## Feature Description

As noticed in https://github.com/google/site-kit-wp/issues/7479#issuecomment-1706366025, our [`Link` component uses many inline SVGs in the CSS](https://github.com/google/site-kit-wp/blob/af3ec820c7885fa724965f81bcbed55c510cae10/assets/sass/components/global/_googlesitekit-cta-link.scss#L52-L143) that cannot be styled when the colour of the text changes/is modified. We want to avoid situations link this where an icon colour does not match the text colour:

![CleanShot 2023-09-05 at 11 39 39](https://github.com/google/site-kit-wp/assets/90871/0b037cb7-1851-4036-b669-680cd55d2030)

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All SVGs for props like `external` in the `Link` component should be changed to use rendered SVGs (similar to https://github.com/google/site-kit-wp/blob/954d8ddd1fa01e09e87a51f8a0e7805d9310f2b6/assets/js/components/KeyMetrics/ChangeMetricsLink.js#L52-L58) instead of CSS SVGs.
* These SVGs should support colour styling so they can match the `currentColor` of the text of the link, so that when the link colour changes (eg. on hover for some components) the icon colour also updates.

## Implementation Brief

* Following classes include Inline svgs:
  * `.googlesitekit-cta-link--back`
  * `.googlesitekit-cta-link--external` 
  * `.googlesitekit-cta-link--arrow`
    * and it's inverse `.googlesitekit-cta-link--inverse` and disabled `.googlesitekit-cta-link--disabled` variations
* Inline svgs should be removed from there, and proper styling applied
* The `external` icon is this one 
![image](https://github.com/google/site-kit-wp/assets/10414808/8036db75-8ee1-43bc-83b2-e41cfbb9c44e)
* The `arrow` icon is this one (Under `Settings -> Connect More Services`)
![image](https://github.com/google/site-kit-wp/assets/10414808/b0889e92-88ab-49b6-8678-f9ee9209f7fa)
* The `back` icon is this one (Shown in dashboard single page view, with non existing url)
![image](https://github.com/google/site-kit-wp/assets/10414808/479516a9-d035-4a8a-a492-f37d223abc95)

* Most icons are available in the icons components list in [Figma](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?type=design&node-id=1669-34953&mode=design&t=cUm6JbmCbJyIWiZs-0)
  * `Back` and `External` icons are in [UI icons](https://www.figma.com/file/ZL8qlbUyR9lmBaF95t0xpq/sitekit-system-gm2%2B?type=design&node-id=1669-15285&mode=design&t=Smd7xOKhXnhhax2N-0) group
* Svg paths should have `fill` attribute set to `currentColor` so it can be controlled via css properties
* `Arrow` icon is not present in the exact shape in Figma (although it looks like `east` icon from `Map` group, but grouped with custom `<circle>`), so it can be used from inline code:
```svg
<svg xmlns="http://www.w3.org/2000/svg" width="13" height="13" viewBox="0 0 13 13">
    <g fill="none" fill-rule="evenodd">
        <circle fill="currentColor" cx="6.5" cy="6.5" r="6.5"/>
        <path d="M3.461 6.96h5.15L6.36 9.21a.464.464 0 0 0 .325.79.459.459 0 0 0 .325-.135l3.037-3.038a.459.459 0 0 0 0-.65L7.015 3.135a.46.46 0 0 0-.65.65L8.61 6.039H3.461a.462.462 0 0 0-.461.46c0 .254.207.462.461.462z" fill="#FFF"/>. 
    </g>
</svg>
```
 * Create two versions of the icon with the provided svg code
    *  `Arrow` default version
       * The circle's `fill` attribute should be set to `currentColor` to be modifiable via CSS (default green color).
       * The arrow retains its specific color - `white`.
    * `ArrowInverse` - inversed version
      * The arrow's `fill` attribute should be set to `currentColor` to be controlled via CSS (default green color)..
      * The circle retains its specific color - `white`.
* In `assets/js/components/Link.js` update `LinkComponent` to include svg versions if `external`, `back` or `arrow` props are passed
* `UserDimensionsPieChart` components should also include `external` svg icon version
* Adjust Existing CSS Tweaks:
  * Search the codebase for `googlesitekit-cta-link--(arrow|external|inverse|disabled|back)`.
  * Ensure that any existing CSS adjustments to the link icons, including usage of `background: none` to hide the external link icon in certain areas, are updated to reflect the new changes.

### Test Coverage

* Update `./stories/tour-tooltips.stories.js` to include svg icons
* Update any failing VRT images

## QA Brief

* The links and CTA's present on the plugin should function correctly and show no regression. 
* The looks of the icon is slightly different due to the difference in Figma icon set and the preciously used inline ones. 

## Changelog entry

* Improve styling of icons associated with links.

**Comment by techanvil** [comment]
Hi @zutigrm, thanks for drafting this IB which is generally looking good. I do have a few points of feedback:

- It looks like there is an additional class with an inline SVGs associated with it which should be included in the IB - [`googlesitekit-cta-link--back`](https://github.com/google/site-kit-wp/blob/5dd5ef7b3041e918d77d66506e608b2f0da8906b/assets/sass/components/global/_googlesitekit-cta-link.scss#L121-L123).
- It would be good to identify which icons from the linked Figma file should be used, as this will a) allow us to review them now, b) save time during implementation, and c) confirm their existence, as if any _don't_ exist we need to figure that out (presumably the thing to do for now would be to simply copy the inlined SVG data to files).
- It's probably worth pointing out that the disabled arrow variant needs to be a variant of the regular arrow as the actual arrow part of it needs to be green.
- Another thing worth noting, to make sure it's not potentially overlooked, is that we should search for `googlesitekit-cta-link--(arrow|external|inverse|disabled)` across the codebase and ensure that any existing CSS tweaks to the link icons are updated as necessary - for example we're using `background: none` to suppress the external link icon in a couple of places.
- You might want to remove your local path component from the path in Test Coverage :)

**Comment by techanvil** [comment]
Thanks @zutigrm! This IB LGTM, although I am going to bump the estimate to 15 as I think that tracking down and ensuring all of the various cases are updated correctly could be rather time consuming in both development and testing. Also, CSS-focused issues can often be more time consuming than they first appear.

**Comment by kuasha420** [comment]
@wpdarren I'm having a look at the `Edit in Analytics` link issue you mentioned in Slack.

**Comment by wpdarren** [comment]
@kuasha420 Should we send this back to Execution do you think?

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@kuasha420 as per our conversation. I have sent this back to Execution. 

> Just noticed a regression in develop branch (I can’t re-create it in the latest release).  There’s no space between ‘Edit’ and ‘in’  :thinking_face:  Looking at the tickets merged, none of them look related, so I am wondering if anyone has any thoughts on what could have caused it. 

![image](https://github.com/google/site-kit-wp/assets/73545194/37dca873-5dd9-48d7-a7d4-1ebbcaf04fef)

**Comment by kuasha420** [comment]
@wpdarren Turns out the issue is not isolated to the Edit links and happens elsewhere where the `VisuallyHidden` component is used. Although the bug originated in this issue, @tofumatt has already created another issue [here](https://github.com/google/site-kit-wp/issues/7968) and proposed a fix in this PoC [PR](https://github.com/google/site-kit-wp/pull/7969). We can note this as a known issue in this ticket, as it's better to be solved in the linked issue. I'm moving this back to QA so you can finish the remaining QA work here. Cheers.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

**Verified:**

@kuasha420 I found a few alignment issues:

1. When setting up a module in Site Kit settings.  The arrow icon isn't aligned with the text. I checked how it looked previously, and it was more aligned with the text. 

Latest release:
![image](https://github.com/google/site-kit-wp/assets/73545194/93b5e5ce-d149-41f1-befd-d1a3c0780c5a)
On Develop branch
![image](https://github.com/google/site-kit-wp/assets/73545194/1bfb61c5-7191-4da3-85e5-d59f4e38ae6a)
 
2. On the entity dashboard, the external link icon does not align with the URL. 

![image](https://github.com/google/site-kit-wp/assets/73545194/e8befce4-bbb0-4b59-867b-56bd13ed3365)

3. On the Dashboard sharing panel, the external link icon does not align with the text. It's only very slight,  but it was more aligned on the latest release than in develop. It looks a pixel out, so the icon is center aligned with the text. 

![image](https://github.com/google/site-kit-wp/assets/73545194/1a552cd5-c93e-4c61-9ae5-db63df437c0c)

**Comment by kuasha420** [comment]
@wpdarren Thank you for your observation. From my testing, all the alignment issues you've mentioned are more or less fixed in the linked [PR](https://github.com/google/site-kit-wp/pull/7969) for #7968. I'm not sure how to proceed here so that we don't duplicate and waste effort, maybe test the failing points once the other issue in QA and passing both together is the way forward?

**Comment by wpdarren** [comment]
QA Update: ⚠️ 

We have several issues that will be fixed #7968:

- Issue with GA and GTM edit links [reported here](https://github.com/google/site-kit-wp/issues/7542#issuecomment-1845082045) 
- Issue with alignments of SVG icons [reported here](https://github.com/google/site-kit-wp/issues/7542#issuecomment-1849909783)

Once the above-mentioned ticket is in QA, I will check that these issues are fixed and approve this.

**Comment by wpdarren** [comment]
### QA Update: ❌ 

@kuasha420 Two of the issues reported above were fixed when 7968 was merged, but I still see some misalignment on the connect services modules.  It's only very slightly off. 

Latest release:
![image](https://github.com/google/site-kit-wp/assets/73545194/7f7adc10-c25b-45a0-8606-0607a897664c)

Main:
![image](https://github.com/google/site-kit-wp/assets/73545194/5c8fcc79-d591-4940-9a10-f506f7fe8edf)

**Comment by aaemnnosttv** [comment]
@wpdarren the alignment changed in #7968 so let's move this feedback to that issue instead.

**Comment by wpdarren** [comment]
@aaemnnosttv this issue existed before 7968 was merged, see [comment above](https://github.com/google/site-kit-wp/issues/7542#issuecomment-1849909783).

**Comment by kuasha420** [comment]
As par Slack discussion, all the issues is getting looked at and fixed by @tofumatt on #7968, So, I'm moving this forward.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

Please see the comment on #7968 for all the tests completed on this issue.  We've completed a thorough check of all of the SVGs and highlighted a number of issues that are now fixed.
